### ISSUE TITLE
Login from CI?

### ISSUE THREAD

**Issue by karanpratapsingh** [cloudflare/cloudflared#477]
is there a way to do `cloudflared login` from CI with let's say cloudflare `api_token`? The idea is to pre-create argo tunnels and use it inside ECS fargate tasks

**Comment by sudarshan-reddy** [comment]
You don’t need a .pen file (so no need to login) to simply run a tunnel. Just pass around the <tunnel is>.json file that got saved when you precreated the tunnel.


https://community.cloudflare.com/t/argo-tunnel-credentials-file/224925

**Comment by karanpratapsingh** [comment]
thanks @sudarshan-reddy but I'll still need to be authenticated when I'll pre-create the tunnel on CI?

**Comment by sudarshan-reddy** [comment]
Yes. Can I suggest you pre-create a tunnel outside of CI and then use that in CI?

**Comment by karanpratapsingh** [comment]
googling around, it seems like people store `cert.pem` file in S3 and pull in CI to use. So trying that. Thank @sudarshan-reddy
